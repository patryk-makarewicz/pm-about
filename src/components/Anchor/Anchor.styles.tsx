import styled, { css } from 'styled-components';
import { Anchor as AntdAnchor } from 'antd';
import { colors } from '../../styles/colors';
import { breakpoint, fontSize, fontWeight } from '../../styles/variables';

type AnchorProps = {
  mobile?: boolean;
};

export const Anchor = styled(AntdAnchor)<AnchorProps>`
  ${({ mobile }) =>
    mobile
      ? css`
          display: flex;
        `
      : css`
          display: none;
        `}

  @media screen and (min-width: ${breakpoint.s}) {
    ${({ mobile }) =>
      mobile
        ? css`
            display: none;
          `
        : css`
            display: flex;
          `}
  }

  .ant-anchor {
    font-size: ${fontSize.m};
    font-weight: ${fontWeight.regular};
    display: flex;
    ${({ mobile }) =>
      mobile &&
      css`
        flex-direction: column;
      `}
  }
  .ant-anchor-ink {
    ${({ mobile }) =>
      mobile &&
      css`
        display: none;
      `}
  }

  .ant-anchor-link-title {
    color: var(--text-color);
    ${({ mobile }) =>
      mobile &&
      css`
        margin-bottom: 20px;
        font-size: ${fontSize.l};
      `}
  }
  .ant-anchor-link-active > .ant-anchor-link-title {
    color: ${colors.green};
  }
`;
