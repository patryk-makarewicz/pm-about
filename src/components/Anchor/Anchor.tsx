import * as Styled from './Anchor.styles';
import { Anchor as AntdAnchor } from 'antd';

const { Link } = AntdAnchor;

type AnchorProps = {
  paths: {
    href: string;
    title: string;
  }[];
  mobile?: boolean;
  onClick?: () => void;
};

export const Anchor = ({ paths, mobile = false, onClick }: AnchorProps) => {
  return (
    <Styled.Anchor affix={false} targetOffset={95} onClick={onClick} mobile={mobile}>
      {paths.map((path) => (
        <Link key={path.href} href={path.href} title={path.title} />
      ))}
    </Styled.Anchor>
  );
};
