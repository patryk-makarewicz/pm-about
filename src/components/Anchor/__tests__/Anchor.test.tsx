import { render, cleanup, queryByText } from '@testing-library/react';

import { Anchor } from '../Anchor';

afterEach(cleanup);

const pathsData = [
  {
    href: '#about',
    title: 'About Testing'
  },
  {
    href: '#tech',
    title: 'Tech Testing'
  }
];

const renderAnchor = ({ mobile = false, paths = pathsData } = {}) => {
  const onClick = jest.fn();
  const { baseElement, container, getByText } = render(<Anchor paths={paths} mobile={mobile} onClick={onClick} />);

  return {
    baseElement,
    onClick,
    container,
    getByText
  };
};

describe('Testing Anchor Component', () => {
  test('should take snapshot with paths', () => {
    const { baseElement } = renderAnchor();

    expect(baseElement).toMatchSnapshot();
  });

  test('should render paths', () => {
    const { getByText } = renderAnchor();

    expect(getByText('About Testing')).toBeTruthy();
  });

  test('should not render paths', () => {
    const { container } = renderAnchor({ paths: [] });

    expect(queryByText(container, 'About Testing')).toBeFalsy();
  });
});
