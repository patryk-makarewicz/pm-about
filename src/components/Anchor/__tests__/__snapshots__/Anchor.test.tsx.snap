// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Anchor Component should take snapshot with paths 1`] = `
.c0 {
  display: none;
}

.c0 .ant-anchor {
  font-size: 1rem;
  font-weight: 400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .ant-anchor-link-title {
  color: var(--text-color);
}

.c0 .ant-anchor-link-active > .ant-anchor-link-title {
  color: #389e0d;
}

@media screen and (min-width:576px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<body>
  <div>
    <div
      class="ant-anchor-wrapper c0"
      style="max-height: 100vh;"
    >
      <div
        class="ant-anchor ant-anchor-fixed"
      >
        <div
          class="ant-anchor-ink"
        >
          <span
            class="ant-anchor-ink-ball"
          />
        </div>
        <div
          class="ant-anchor-link"
        >
          <a
            class="ant-anchor-link-title"
            href="#about"
            title="About Testing"
          >
            About Testing
          </a>
        </div>
        <div
          class="ant-anchor-link"
        >
          <a
            class="ant-anchor-link-title"
            href="#tech"
            title="Tech Testing"
          >
            Tech Testing
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
